### DISCUSSION TITLE
Correct data transfer mqtty into variable?

### DISCUSSION THREAD

**Discussion by HofeBY** [arendst/Tasmota#20551] [Q&A]
I use TM13.3.0.2 on ESP32 with an SML read head.
The measured value 2.8.0 delivery current is therefore accurate to 0,1Wh
correctly transferred via mqtty to my IOBroker with the value 175784,0099 kWh.
That works very well.

On this ESP32 component I read the script section ">T".

"A280=AMT681#A2_8_0" from mqtty source into a variable.
where A280 is my variable name used.
AMT681 is the name of my meter used. A first meter, MT681 meter type
A2_8_0 means "A" first counter, "2_8_0" means fed-in excess electricity in kWh with an accuracy of 0,1Wh

But now it is the case that this reading only takes place with 2 places after the decimal point
which in this case results in only 175784,0000 kWh as the value in the variable "A280"
land for further processing. So 9,9Wh is too little.

How can I get the data transfer precision from mqtty to a local variable?
currently expand 2 decimal places to 4 decimal places?
Ideally there would be a way to transfer the same value as the one in the mqtty
could be brought into the variable.
I don't understand why this is arbitrary is limited to 2 decimal places.

Thank you for your support

HofeBY

**Comment by gemu2015** [comment]
scripter variables are always float precision internally.  if you print a variable the default number of decimals is set to 2.
you may set the printing number precision to 4 with dp(4)

**Reply by HofeBY** [reply]
Hello gemu2015

I know that the output is entered by default if nothing else
is represented with two decimal places.

That's why I use the line when outputting the variable "A280" in section ">W".
Meter reading A280 Supply energy: {m} %4A280% kWh
and then it says 175784.0000 kWh and the correct 175784.0099 kWh was recorded in IOBroker.
And I already described that the correct value is in the console LOG.

The error occurs when the data is transferred from mqtty to the variable.
There is no problem with the representation of the variable content.

Thank you for your support, but it´s not a problem of displaying the variable

HofeBY

**Reply by gemu2015** [reply]
you should better get the value directly from SML by sml[x] instead
 of reading MQTT output.  you can use sml[x] in any part of the script not only in >T  (see docs)

**Reply by HofeBY** [reply]
can you show an example to do this ?
maybe another way to read in values will work.
i read three Meters.

**Reply by gemu2015** [reply]
sml[x] where x is the decode line to read independent of the meter
